internal/channel/channel.go

```diff
@@ -20,109 +19,149 @@ type Channel struct {
 	Config string `db:"config" json:"-"` // excluded from JSON config dump as this may contain sensitive information
 
 	Logger *zap.SugaredLogger
-	cmd    *exec.Cmd
-	rpc    *rpc.RPC
-	mu     sync.Mutex
-}
 
-func (c *Channel) Start(pluginDir string) error {
-	c.mu.Lock()
-	defer c.mu.Unlock()
+	newConfigCh chan string
+	pluginCh    chan *Plugin
 
-	if c.cmd != nil && c.rpc.Err() == nil {
-		return nil
-	}
+	pluginCtx       context.Context
+	pluginCtxCancel func()
+}
 
-	if matched, _ := regexp.MatchString("^[a-zA-Z0-9]*$", c.Type); !matched {
-		return fmt.Errorf("channel type must only contain a-zA-Z0-9, %q given", c.Type)
-	}
+// Start initializes the channel and starts the plugin in the background
+func (c *Channel) Start(logger *zap.SugaredLogger, ctx context.Context) {
+	c.Logger = logger
+	c.newConfigCh = make(chan string)
+	c.pluginCh = make(chan *Plugin)
+	c.pluginCtx, c.pluginCtxCancel = context.WithCancel(ctx)
 
-	path := filepath.Join(pluginDir, c.Type)
+	go c.runPlugin(c.Config)
+}
 
-	cmd := exec.Command(path)
+// initPlugin returns a new Plugin or nil if an error occurred during initialization
+func (c *Channel) initPlugin(config string) *Plugin {
+	c.Logger.Debug("Initializing channel plugin")
 
-	writer, err := cmd.StdinPipe()
+	p, err := NewPlugin(c.Type, c.Logger)
```

Comment:
Actually, `c.Type` has the same issue [that I mentioned earlier for `c.Config`](https://github.com/Icinga/icinga-notifications/pull/108#discussion_r1380093296).